### PR TITLE
Refactored method preprocess_request in Flask.

### PR DESCRIPTION
**Pull request by AvivC**
Code duplication was eliminated. The docstring was also improved.

### CODE REVIEW

**Comment by davidism** [comment]
I'm on board with improving the docs, but I don't see any reason this needs to be split into four functions.

**Comment by AvivC** [comment]
@davidism Agree. I removed the two extra functions. The PR now simply eliminates code duplication.

**Comment by davidism** [comment]
Two for loops is not code duplication, and this function is unlikely to change.  Documentation only is fine.

File: flask/app.py [hunk 1]
```diff
@@ -1791,32 +1791,39 @@ def handle_url_build_error(self, error, endpoint, values):
...
-        This also triggers the :meth:`url_value_processor` functions before
-        the actual :meth:`before_request` functions are called.
+        If any of the before_request_funcs returns a value, it's handled as if it was
+        the return value from the view function, and further request handling is stopped.
         """
-        bp = _request_ctx_stack.top.request.blueprint
-
-        funcs = self.url_value_preprocessors.get(None, ())
-        if bp is not None and bp in self.url_value_preprocessors:
-            funcs = chain(funcs, self.url_value_preprocessors[bp])
```
Comment (AvivC) [inline flask/app.py]: @davidism The duplication I was referring to isn't the for loops, but the three lines of code that come before each one.
They are exactly identical except for the variables, so IMO it makes sense to factor them out to a function.

**Comment by davidism** [comment]
Thanks, but I don't think this needs to be refactored.  It doesn't improve anything.  If you'd like to resubmit the documentation clarification, that's fine.

**Comment by AvivC** [comment]
I will :)

**Pull request closed without merging**